### PR TITLE
fix artisan serve command crash on windows

### PR DESCRIPTION
**Pull request by rajatpatelz**
I have tried a workaround to fix the artisan serve command crash issue on windows.
Issues : #43673, #43653

**Fix:** Implemented condition to check if the request has valid port no and date, or else ignore the request ( not add in **$requestpool** and hence request details will not be visible in console output).

### CODE REVIEW

**Comment by taylorotwell** [comment]
Can you make the diff much simpler by just bailing early if we have invalid date or port?

File: src/Illuminate/Foundation/Console/ServeCommand.php [hunk 1]
```diff
@@ -227,24 +227,22 @@ protected function canTryAnotherPort()
...
                     false,
                 ];
-            } elseif (str($line)->contains([' [200]: GET '])) {
-                $requestPort = $this->getRequestPortFromLine($line);
-
+            } elseif (str($line)->contains([' [200]: GET ']) && null !== $requestPort) {
                 $this->requestsPool[$requestPort][1] = trim(explode('[200]: GET', $line)[1]);
-            } elseif (str($line)->contains(' Closing')) {
-                $requestPort = $this->getRequestPortFromLine($line);
+            } elseif (str($line)->contains(' Closing') && null !== $requestPort && isset($this->requestsPool[$requestPort]) && count($this->requestsPool[$requestPort]) == 2) {
```
Comment (taylorotwell) [inline src/Illuminate/Foundation/Console/ServeCommand.php:245]: What is this line actually checking? Why the hard-coded 2?

Comment (rajatpatelz) [reply src/Illuminate/Foundation/Console/ServeCommand.php:245]: **$requestsPool** variable stores the **Start Time** and **filename** of the handled requests. In this line, we are checking if both start time and file name are present in the $requestsPool[$requestPort] variable, hence the hard-coded value 2.

**Comment by nunomaduro** [comment]
Can you elaborate on what makes the "$line" not have a valid port or date?

**Comment by rajatpatelz** [comment]
> @rajatpatelz Can you elaborate on what makes the "$line" or request not have a valid port or date?

I am not sure why this is happening. Sometimes "$line" does not contain either a complete date or a valid port number.

```
INFO  Server running on [http://127.0.0.1:8000].

  Press Ctrl+C to stop the server

  2022-08-12 16:33:19 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:33:20 /css/filament.css ................................................................................................................................................. ~ 0s
  2022-08-12 16:33:20 /storage/uploads/pat_logo/BPb47pYtiiadug54ymghhKKyOCqiw3-metab3JnLWxvZ28uanBn-.jpg ............................................................................ ~ 0s
  2022-08-12 16:33:20 /storage/uploads/profile/4lLeCIodq7YXje6WyXhSVZV1Bkqjk5-metaMDc1LUFuYW5kIEtvdGVjaGEuanBn-.jpg .............................................................. ~ 0s
  2022-08-12 16:33:20 /js/jquery-3.6.0.min.js ........................................................................................................................................... ~ 0s
  2022-08-12 16:33:20 /storage/uploads/profile/YCzbQUvC3cVJZeXM66GF07ALwZxilR-metaMDAyLWFiaGlzaGVrIG1hcm90aGkuanBn-.jpg .......................................................... ~ 0s
  2022-08-12 16:33:20 /js/jquery.plugin.min.js .......................................................................................................................................... ~ 0s
  2022-08-12 16:33:20 /js/app.js ........................................................................................................................................................ ~ 0s
  2022-08-12 16:33:20 /js/jquery.countdown.min.js ....................................................................................................................................... ~ 0s
  2022-08-12 16:33:20 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:33:20 /storage/uploads/profile/default.jpg ....................................................................................................................... ~ 0s
  2022-08-12 16:33:20 /storage/uploads/tlogo/MtM3O1sGmvnYVge1yiMGjE8uwNWhld-metaMS5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:20 /storage/uploads/tlogo/yMoy30sgKs2Mfe9a6WPJB70jpBAA1K-metaMi5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:20 /storage/uploads/tlogo/0mNPASq0bnLeZYucdow7VMTbmxUTcx-metaMy5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:20 /storage/uploads/tlogo/RI8BGaCZ78Mtei1mrNYxMd1JHi06ji-metaNC5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:21 /countdown2.wav ................................................................................................................................................... ~ 0s
  2022-08-12 16:33:27 ................................................................................................................................................................... ~ 2s
  2022-08-12 16:33:29 /css/filament.css ................................................................................................................................................. ~ 0s
  2022-08-12 16:33:29 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
   WARN  /yMoy30sgKs2Mfe9a6WPJB70jpBAA1K-metaMi5qcGc=-.jpg.

  2022-08-12 16:33:29 /storage/uploads/tlogo/MtM3O1sGmvnYVge1yiMGjE8uwNWhld-metaMS5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:29 /storage/uploads/tlogo ........................................................................................................................................ ~ 0s
  2022-08-12 16:33:29 /storage/uploads/tlogo/0mNPASq0bnLeZYucdow7VMTbmxUTcx-metaMy5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:29 /storage/uploads/tlogo/RI8BGaCZ78Mtei1mrNYxMd1JHi06ji-metaNC5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:31 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:33:40 ................................................................................................................................................................... ~ 5s
  2022-08-12 16:33:52 ................................................................................................................................................................... ~ 4s
  2022-08-12 16:33:56 /css/app.css ...................................................................................................................................................... ~ 0s
  2022-08-12 16:33:56 /js/jquery-3.6.0.min.js ........................................................................................................................................... ~ 0s
  2022-08-12 16:33:56 /js/app.js ........................................................................................................................................................ ~ 0s
  2022-08-12 16:33:57 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:33:57 /storage/uploads/pat_banner/RaZpL1JGnVAysRa0YLziplUjBSsLz9-metab3JnLWxvZ28uanBn-.jpg .......................................................................... ~ 0s
  2022-08-12 16:33:58 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:33:59 /storage/uploads/profile/default.jpg ....................................................................................................................... ~ 0s
  2022-08-12 16:33:59 /storage/uploads/tlogo/0mNPASq0bnLeZYucdow7VMTbmxUTcx-metaMy5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:59 /storage/uploads/tlogo/MtM3O1sGmvnYVge1yiMGjE8uwNWhld-metaMS5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:59 /storage/uploads/tlogo/yMoy30sgKs2Mfe9a6WPJB70jpBAA1K-metaMi5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:33:59 /storage/uploads/tlogo/RI8BGaCZ78Mtei1mrNYxMd1JHi06ji-metaNC5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:34:06 ................................................................................................................................................................... ~ 0s
  2022-08-12 16:34:06 ................................................................................................................................................................... ~ 0s
  2022-08-12 16:34:06 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:34:06 /css/app.css ...................................................................................................................................................... ~ 1s
   WARN  [Fri Aug 12 16:34:07 202.

  2022-08-12 16:34:06 /js/jquery-3.6.0.min.js ........................................................................................................................................... ~ 1s
  2022-08-12 16:34:07 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:34:07 /storage/uploads/pat_banner/RaZpL1JGnVAysRa0YLziplUjBSsLz9-metab3JnLWxvZ28uanBn-.jpg .......................................................................... ~ 0s
  2022-08-12 16:34:06 /js/app.js ........................................................................................................................................................ ~ 1s
  2022-08-12 16:34:08 ................................................................................................................................................................... ~ 0s
  2022-08-12 16:34:08 /favicon.ico ...................................................................................................................................................... ~ 0s
  2022-08-12 16:34:08 /storage/uploads/profile/default.jpg ....................................................................................................................... ~ 0s
  2022-08-12 16:34:08 /storage/uploads/tlogo/0mNPASq0bnLeZYucdow7VMTbmxUTcx-metaMy5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:34:08 /storage/uploads/tlogo/MtM3O1sGmvnYVge1yiMGjE8uwNWhld-metaMS5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:34:08 /storage/uploads/tlogo/yMoy30sgKs2Mfe9a6WPJB70jpBAA1K-metaMi5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:34:08 /storage/uploads/tlogo/RI8BGaCZ78Mtei1mrNYxMd1JHi06ji-metaNC5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:37:21 ................................................................................................................................................................... ~ 5s
  2022-08-12 16:37:45 ................................................................................................................................................................... ~ 2s
  2022-08-12 16:37:47 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:37:47 /js/jquery-3.6.0.min.js ........................................................................................................................................... ~ 0s
  2022-08-12 16:37:47 /css/app.css ...................................................................................................................................................... ~ 0s
  2022-08-12 16:37:47 /storage/uploads/pat_banner/RaZpL1JGnVAysRa0YLziplUjBSsLz9-metab3JnLWxvZ28uanBn-.jpg .......................................................................... ~ 0s
  2022-08-12 16:37:47 /js/app.js ........................................................................................................................................................ ~ 0s
  2022-08-12 16:37:48 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:37:49 /storage/uploads/profile/default.jpg ....................................................................................................................... ~ 0s
  2022-08-12 16:37:49 /storage/uploads/tlogo/0mNPASq0bnLeZYucdow7VMTbmxUTcx-metaMy5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:37:49 /storage/uploads/tlogo/MtM3O1sGmvnYVge1yiMGjE8uwNWhld-metaMS5qcGc=-.jpg ....................................................................................... ~ 0s
   WARN  e/uploads/tlogo/yMoy30sgKs2Mfe9a6WPJB70jpBAA1K-metaMi5qcGc=-.jpg.

  2022-08-12 16:37:49 /storag ........................................................................................................................................................... ~ 0s
  2022-08-12 16:37:49 /storage/uploads/tlogo/RI8BGaCZ78Mtei1mrNYxMd1JHi06ji-metaNC5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:37:55 ................................................................................................................................................................... ~ 2s
  2022-08-12 16:37:57 /storage/uploads/pat_logo/BPb47pYtiiadug54ymghhKKyOCqiw3-metab3JnLWxvZ28uanBn-.jpg ............................................................................ ~ 0s
  2022-08-12 16:37:57 /js/jquery-3.6.0.min.js ........................................................................................................................................... ~ 0s
  2022-08-12 16:37:57 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:37:57 /css/app.css ...................................................................................................................................................... ~ 0s
  2022-08-12 16:37:57 /storage/uploads/pat_logo/btQy6u56lPMwVscidn8t8p4KHvWWrK-metabXBsLWxvZ28tai5qcGc=-.jpg ........................................................................ ~ 0s
  2022-08-12 16:37:57 /js/app.js ........................................................................................................................................................ ~ 0s
  2022-08-12 16:38:00 ................................................................................................................................................................... ~ 2s
  2022-08-12 16:38:02 /css/app.css ...................................................................................................................................................... ~ 0s
  2022-08-12 16:38:02 /js/jquery-3.6.0.min.js ........................................................................................................................................... ~ 0s
  2022-08-12 16:38:02 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:38:02 /storage/uploads/pat_banner/RaZpL1JGnVAysRa0YLziplUjBSsLz9-metab3JnLWxvZ28uanBn-.jpg .......................................................................... ~ 0s
  2022-08-12 16:38:02 /js/app.js ........................................................................................................................................................ ~ 0s
  2022-08-12 16:38:02 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:38:03 /storage/uploads/profile/default.jpg ....................................................................................................................... ~ 0s
  2022-08-12 16:38:03 /storage/uploads/tlogo/0mNPASq0bnLeZYucdow7VMTbmxUTcx-metaMy5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:38:03 /storage/uploads/tlogo/MtM3O1sGmvnYVge1yiMGjE8uwNWhld-metaMS5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:38:03 /storage/uploads/tlogo/yMoy30sgKs2Mfe9a6WPJB70jpBAA1K-metaMi5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:38:03 /storage/uploads/tlogo/RI8BGaCZ78Mtei1mrNYxMd1JHi06ji-metaNC5qcGc=-.jpg ....................................................................................... ~ 0s
   WARN  reconnection.

  2022-08-12 16:38:05 ................................................................................................................................................................... ~ 0s
  2022-08-12 16:38:05 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:38:06 /css/app.css ...................................................................................................................................................... ~ 0s
  2022-08-12 16:38:06 /js/app.js ........................................................................................................................................................ ~ 0s
  2022-08-12 16:38:07 /storage/uploads/pat_banner/RaZpL1JGnVAysRa0YLziplUjBSsLz9-metab3JnLWxvZ28uanBn-.jpg .......................................................................... ~ 0s
  2022-08-12 16:38:07 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:38:07 /favicon.ico ...................................................................................................................................................... ~ 0s
  2022-08-12 16:38:19 ................................................................................................................................................................... ~ 0s
  2022-08-12 16:38:19 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:38:20 ................................................................................................................................................................... ~ 0s
  2022-08-12 16:38:20 /css/app.css ...................................................................................................................................................... ~ 0s
  2022-08-12 16:38:20 /js/jquery-3.6.0.min.js ........................................................................................................................................... ~ 0s
  2022-08-12 16:38:20 /storage/uploads/pat_banner/RaZpL1JGnVAysRa0YLziplUjBSsLz9-metab3JnLWxvZ28uanBn-.jpg .......................................................................... ~ 0s
  2022-08-12 16:38:20 /imgs/pat_logo.png ................................................................................................................................................ ~ 0s
  2022-08-12 16:38:20 /js/app.js ........................................................................................................................................................ ~ 0s
  2022-08-12 16:38:21 ................................................................................................................................................................... ~ 1s
  2022-08-12 16:38:22 /favicon.ico ...................................................................................................................................................... ~ 0s
  2022-08-12 16:38:22 /storage/uploads/profile/default.jpg ....................................................................................................................... ~ 0s
  2022-08-12 16:38:22 /storage/uploads/tlogo/0mNPASq0bnLeZYucdow7VMTbmxUTcx-metaMy5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:38:22 /storage/uploads/tlogo/MtM3O1sGmvnYVge1yiMGjE8uwNWhld-metaMS5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:38:22 /storage/uploads/tlogo/yMoy30sgKs2Mfe9a6WPJB70jpBAA1K-metaMi5qcGc=-.jpg ....................................................................................... ~ 0s
  2022-08-12 16:38:22 /storage/uploads/tlogo/RI8BGaCZ78Mtei1mrNYxMd1JHi06ji-metaNC5qcGc=-.jpg ....................................................................................... ~ 0s
   WARN  [Fri Aug 12 16:38:29 20.

  2022-08-12 16:38:29
   ErrorException

  Undefined array key 1

  at E:\xampp\htdocs\pat\vendor\laravel\framework\src\Illuminate\Foundation\Console\ServeCommand.php:289
    285▕     protected function getDateFromLine($line)
    286▕     {
    287▕         preg_match('/^\[([^\]]+)\]/', $line, $matches);
    288▕
  ➜ 289▕         return Carbon::createFromFormat('D M d H:i:s Y', $matches[1]);
    290▕     }
    291▕
    292▕     /**
    293▕      * Get the request port from the given PHP server output.
PHP server output.

  1   E:\xampp\htdocs\pat\vendor\laravel\framework\src\Illuminate\Foundation\Console\ServeCommand.php:289
      Illuminate\Foundation\Bootstrap\HandleExceptions::Illuminate\Foundation\Bootstrap\{closure}("Undefined array key 1", "E:\xampp\htdocs\pat\vendor\laravel\framework\src\Illuminate\Foun
dation\Console\ServeCommand.php")

  2   E:\xampp\htdocs\pat\vendor\laravel\framework\src\Illuminate\Foundation\Console\ServeCommand.php:260
      Illuminate\Foundation\Console\ServeCommand::getDateFromLine("22] 127.0.0.1:57100 Closing")  
```

This is output of serve command, as you can see in the last line, "$line" does not contain a valid date time and this is causing the serve command to crash.

**Comment by nunomaduro** [comment]
> I am not sure why this is happening. Sometimes "$line" does not contain either a complete date or a valid port number.

Seems that is an issue with Symfony's process when running in your environment. Can you try to tweak the Symfony process instead, so this don't happen with the process output?

**Comment by inxty** [comment]
@nunomaduro I am receiving same error in Laravel 10, and it is a random error.
What do you mean by Symfony's process?

**Pull request closed without merging**